LPC Projects/PWM_Sample/src/PWM_Sample.c

```diff
@@ -11,42 +11,225 @@
 #include "board.h"
 #include <stdio.h>
 
-// TODO: insert other include files here
+/* GPIO pins for encoder interrupt */
+#define PIN_A					2 /* GPIO pin number mapped to PIN_A */
+#define PIN_A_PORT				0 /* GPIO port number mapped to PIN_A */
+#define PIN_B					5 /* GPIO pin number mapped to PIN_B */
+#define PIN_B_PORT				0 /* GPIO port number mapped to PIN_B */
+#define IOCON_PIN_A_ID			IOCON_PIO0_2 /* IOCON pin identifer */
+#define IOCON_PIN_B_ID			IOCON_PIO0_5 /* IOCON pin identifer */
+#define ENCODER_IRQ_HANDLER		PIOINT0_IRQHandler /* GPIO interrupt IRQ function name */
+#define ENCODER_NVIC_NAME		EINT0_IRQn /* GPIO interrupt NVIC interrupt name */
 
-// TODO: insert other definitions and declarations here
-///**
-// * @brief	Handle interrupt from SysTick timer
-// * @return	Nothing
-// */
-void SysTick_Handler(void)
+#include <cr_section_macros.h>
+
+volatile uint8_t last_state;
+volatile long ticks;
+
+volatile int desired_steering_angle = 0;
+//volatile int direction = 1; // +1 forwards, -1 backwards
+
+int danger = 0;
+
+
+#define TEST_CCAN_BAUD_RATE 1000000
+CCAN_MSG_OBJ_T msg_obj;
+
+void baudrateCalculate(uint32_t baud_rate, uint32_t *can_api_timing_cfg)
 {
-	Board_LED_Set(0, false);
+	uint32_t pClk, div, quanta, segs, seg1, seg2, clk_per_bit, can_sjw;
+	Chip_Clock_EnablePeriphClock(SYSCTL_CLOCK_CAN);
+	pClk = Chip_Clock_GetMainClockRate();
+
+	clk_per_bit = pClk / baud_rate;
+
+	for (div = 0; div <= 15; div++) {
+		for (quanta = 1; quanta <= 32; quanta++) {
+			for (segs = 3; segs <= 17; segs++) {
+				if (clk_per_bit == (segs * quanta * (div + 1))) {
+					segs -= 3;
+					seg1 = segs / 2;
+					seg2 = segs - seg1;
+					can_sjw = seg1 > 3 ? 3 : seg1;
+					can_api_timing_cfg[0] = div;
+					can_api_timing_cfg[1] =
+						((quanta - 1) & 0x3F) | (can_sjw & 0x03) << 6 | (seg1 & 0x0F) << 8 | (seg2 & 0x07) << 12;
+					return;
+				}
+			}
+		}
+	}
+}
+
+/*	CAN receive callback */
+/*	Function is executed by the Callback handler after
+    a CAN message has been received */
+void CAN_rx(uint8_t msg_obj_num) {
+	/* Determine which CAN message has been received */
+	msg_obj.msgobj = msg_obj_num;
+	/* Now load up the msg_obj structure with the CAN message */
+	LPC_CCAN_API->can_receive(&msg_obj);
+	int new_desired_angle = 0;
+
+//	if (msg_obj_num == 1)
```

Comment:
Is this block of dead code from when we were determining how to have multiple CAN message types?  Remove if this was solved by following block of not-dead code